### ISSUE TITLE
Help with package dependencies

### ISSUE THREAD

**Issue by thanksforyourworks** [annegnx/PnP-Flow#3]
I'm sorry to bother you again on vacation, but I'm having some problems running the program. If you could provide the version of python you are using and the specific version in the requirements.txt file it would be very helpful in troubleshooting the problem. I would appreciate it if you could reply again!

**Comment by SegoleneMartin** [comment]
Hi, are you still having issues running the code? Here is the list of packages installed in my conda environment. Not everything is necessary but hope it helps.

# This file may be used to create an environment using:
# $ conda create --name <env> --file <this file>
# platform: linux-64
_libgcc_mutex=0.1=main
_openmp_mutex=5.1=1_gnu
_sysroot_linux-64_curr_repodata_hack=3=haa98f57_10
absl-py=2.1.0=pypi_0
aom=3.6.0=h6a678d5_0
archspec=0.2.3=pyhd3eb1b0_0
beautifulsoup4=4.12.3=pypi_0
binutils_impl_linux-64=2.38=h2a08ee3_1
binutils_linux-64=2.38.0=hc2dff05_0
blas=1.0=mkl
bleach=6.1.0=pypi_0
blosc=1.21.3=h6a678d5_0
boltons=23.0.0=py312h06a4308_0
brotli=1.0.9=h5eee18b_8
brotli-bin=1.0.9=h5eee18b_8
brotli-python=1.0.9=py312h6a678d5_8
brunsli=0.1=h2531618_0
bzip2=1.0.8=h5eee18b_6
c-ares=1.19.1=h5eee18b_0
ca-certificates=2024.8.30=hbcca054_0
certifi=2024.7.4=py312h06a4308_0
cffi=1.16.0=py312h5eee18b_1
cfitsio=3.470=h5893167_7
charls=2.2.0=h2531618_0
charset-normalizer=2.0.4=pyhd3eb1b0_0
click=8.1.7=pypi_0
conda-content-trust=0.2.0=py312h06a4308_1
conda-package-handling=2.2.0=py312h06a4308_1
conda-package-streaming=0.9.0=py312h06a4308_0
contextlib2=21.6.0=pypi_0
contourpy=1.2.0=py312hdb19cb5_0
cryptography=42.0.5=py312hdda0065_1
cuda-cudart=11.8.89=0
cuda-cupti=11.8.87=0
cuda-libraries=11.8.0=0
cuda-nvrtc=11.8.89=0
cuda-nvtx=11.8.86=0
cuda-opencl=12.5.39=0
cuda-runtime=11.8.0=0
cuda-version=12.5=3
cudatoolkit-dev=11.7.0=h1de0b5d_6
cycler=0.11.0=pyhd3eb1b0_0
cyrus-sasl=2.1.28=h52b45da_1
dav1d=1.2.1=h5eee18b_0
dbus=1.13.18=hb2f20db_0
deepinv=0.2.1=pypi_0
distro=1.9.0=py312h06a4308_0
docker-pycreds=0.4.0=pypi_0
einops=0.8.0=pypi_0
expat=2.6.2=h6a678d5_0
ffmpeg=4.3=hf484d3e_0
filelock=3.13.1=py312h06a4308_0
fmt=9.1.0=hdb19cb5_1
fontconfig=2.14.1=h4c34cd2_2
fonttools=4.51.0=py312h5eee18b_0
freetype=2.12.1=h4a9f257_0
freia=0.2=pypi_0
fsspec=2024.6.0=pypi_0
gcc_impl_linux-64=11.2.0=h1234567_1
gcc_linux-64=11.2.0=h5c386dc_0
gdown=5.2.0=pypi_0
giflib=5.2.1=h5eee18b_3
gitdb=4.0.11=pypi_0
gitpython=3.1.43=pypi_0
glib=2.78.4=h6a678d5_0
glib-tools=2.78.4=h6a678d5_0
gmp=6.2.1=h295c915_3
gnutls=3.6.15=he1e5248_0
gst-plugins-base=1.14.1=h6a678d5_1
gstreamer=1.14.1=h5eee18b_1
gxx_impl_linux-64=11.2.0=h1234567_1
gxx_linux-64=11.2.0=hc2dff05_0
h5py=3.12.1=pypi_0
hdf5storage=0.1.19=pypi_0
icu=73.1=h6a678d5_0
idna=3.7=py312h06a4308_0
ignite=1.1.0=pypi_0
imagecodecs=2023.1.23=py312h81b8100_1
imageio=2.33.1=py312h06a4308_0
intel-openmp=2023.1.0=hdb19cb5_46306
jax=0.4.31=pypi_0
jaxlib=0.4.31=pypi_0
jaxtyping=0.2.19=pyhd8ed1ab_0
jinja2=3.1.4=py312h06a4308_0
joblib=1.4.0=py312h06a4308_0
jpeg=9e=h5eee18b_1
jsonpatch=1.33=py312h06a4308_1
jsonpointer=2.1=pyhd3eb1b0_0
jxrlib=1.1=h7b6447c_2
kaggle=1.6.17=pypi_0
keopscore=2.2.3=pypi_0
kernel-headers_linux-64=3.10.0=h57e8cba_10
kiwisolver=1.4.4=py312h6a678d5_0
krb5=1.20.1=h143b758_1
lame=3.100=h7b6447c_0
lazy_loader=0.3=py312h06a4308_0
lcms2=2.12=h3be6417_0
ld_impl_linux-64=2.38=h1181459_1
lerc=3.0=h295c915_0
libaec=1.0.4=he6710b0_1
libarchive=3.6.2=h6ac8c49_3
libavif=0.11.1=h5eee18b_0
libbrotlicommon=1.0.9=h5eee18b_8
libbrotlidec=1.0.9=h5eee18b_8
libbrotlienc=1.0.9=h5eee18b_8
libclang=14.0.6=default_hc6dbbc7_1
libclang13=14.0.6=default_he11475f_1
libcublas=11.11.3.6=0
libcufft=10.9.0.58=0
libcufile=1.10.0.4=0
libcups=2.4.2=h2d74bed_1
libcurand=10.3.6.39=0
libcurl=8.7.1=h251f7ec_0
libcusolver=11.4.1.48=0
libcusparse=11.7.5.86=0
libdeflate=1.17=h5eee18b_1
libedit=3.1.20230828=h5eee18b_0
libev=4.33=h7f8727e_1
libffi=3.4.4=h6a678d5_1
libgcc-devel_linux-64=11.2.0=h1234567_1
libgcc-ng=11.2.0=h1234567_1
libgfortran-ng=11.2.0=h00389a5_1
libgfortran5=11.2.0=h1234567_1
libglib=2.78.4=hdc74915_0
libgomp=11.2.0=h1234567_1
libiconv=1.16=h5eee18b_3
libidn2=2.3.4=h5eee18b_0
libjpeg-turbo=2.0.0=h9bf148f_0
libllvm14=14.0.6=hdb19cb5_3
libmamba=1.5.8=hfe524e5_2
libmambapy=1.5.8=py312h2dafd23_2
libnghttp2=1.57.0=h2d74bed_0
libnpp=11.8.0.86=0
libnvjitlink=12.1.105=0
libnvjpeg=11.9.0.86=0
libpng=1.6.39=h5eee18b_0
libpq=12.17=hdbd6064_0
libsolv=0.7.24=he621ea3_1
libssh2=1.11.0=h251f7ec_0
libstdcxx-devel_linux-64=11.2.0=h1234567_1
libstdcxx-ng=11.2.0=h1234567_1
libtasn1=4.19.0=h5eee18b_0
libtiff=4.5.1=h6a678d5_0
libunistring=0.9.10=h27cfd23_0
libuuid=1.41.5=h5eee18b_0
libwebp-base=1.3.2=h5eee18b_0
libxcb=1.15=h7f8727e_0
libxkbcommon=1.0.1=h5eee18b_1
libxml2=2.10.4=hfdd30dd_2
libzopfli=1.0.3=he6710b0_0
lightning-utilities=0.11.6=pypi_0
linear_operator=0.5.3=py_0
llvm-openmp=14.0.6=h9e868ea_0
lpips=0.1.4=pypi_0
lz4-c=1.9.4=h6a678d5_1
markupsafe=2.1.3=py312h5eee18b_0
matplotlib=3.8.4=py312h06a4308_0
matplotlib-base=3.8.4=py312h526ad5a_0
menuinst=2.0.2=py312h06a4308_1
mkl=2023.1.0=h213fc3f_46344
mkl-service=2.4.0=py312h5eee18b_1
mkl_fft=1.3.8=py312h5eee18b_0
mkl_random=1.2.4=py312hdb19cb5_0
ml-collections=0.1.1=pypi_0
ml-dtypes=0.4.0=pypi_0
mpmath=1.3.0=py312h06a4308_0
mysql=5.7.24=h721c034_2
ncurses=6.4=h6a678d5_0
nettle=3.7.3=hbbd107a_1
networkx=3.2.1=py312h06a4308_0
ninja=1.11.0=h924138e_0
numpy=1.26.0=pypi_0
opencv-python=4.10.0.84=pypi_0
openh264=2.1.1=h4ff587b_0
openjpeg=2.4.0=h3ad879b_0
openssl=3.0.14=h5eee18b_0
opt-einsum=3.3.0=pypi_0
packaging=23.2=py312h06a4308_0
pandas=2.2.2=pypi_0
pcre2=10.42=hebb0a14_1
pillow=10.3.0=py312h5eee18b_0
pip=24.0=py312h06a4308_0
platformdirs=3.10.0=py312h06a4308_0
pluggy=1.0.0=py312h06a4308_1
ply=3.11=py312h06a4308_1
pot=0.9.3=pypi_0
protobuf=5.28.2=pypi_0
psutil=6.0.0=pypi_0
pybind11=2.12.0=pypi_0
pybind11-abi=5=hd3eb1b0_0
pycosat=0.6.6=py312h5eee18b_1
pycparser=2.21=pyhd3eb1b0_0
pykeops=2.2.3=pypi_0
pyparsing=3.0.9=py312h06a4308_0
pyqt=5.15.10=py312h6a678d5_0
pyqt5-sip=12.13.0=py312h5eee18b_0
pysocks=1.7.1=py312h06a4308_0
python=3.12.3=h996f2a0_1
python-dateutil=2.9.0post0=py312h06a4308_2
python-slugify=8.0.4=pypi_0
pytorch=2.4.0=py3.12_cuda11.8_cudnn9.1.0_0
pytorch-cuda=11.8=h7e8668a_5
pytorch-ignite=0.5.1=pypi_0
pytorch-mutex=1.0=cuda
pytz=2024.1=pypi_0
pyyaml=6.0.1=py312h5eee18b_0
qt-main=5.15.2=h53bd1ea_10
readline=8.2=h5eee18b_0
reproc=14.2.4=h6a678d5_2
reproc-cpp=14.2.4=h6a678d5_2
requests=2.31.0=py312h06a4308_1
ruamel.yaml=0.17.21=py312h5eee18b_0
scikit-image=0.22.0=py312h526ad5a_0
scikit-learn=1.4.2=py312h526ad5a_1
scipy=1.13.1=py312hc5e2394_0
sentry-sdk=2.14.0=pypi_0
setproctitle=1.3.3=pypi_0
setuptools=69.5.1=py312h06a4308_0
sip=6.7.12=py312h6a678d5_0
six=1.16.0=pyhd3eb1b0_1
smmap=5.0.1=pypi_0
snappy=1.1.10=h6a678d5_1
soupsieve=2.5=pypi_0
sqlite=3.45.3=h5eee18b_0
sympy=1.12=py312h06a4308_0
sysroot_linux-64=2.17=h57e8cba_10
tbb=2021.8.0=hdb19cb5_0
text-unidecode=1.3=pypi_0
threadpoolctl=2.2.0=pyh0d69192_0
tifffile=2023.4.12=py312h06a4308_0
tk=8.6.14=h39e8969_0
torchaudio=2.4.0=py312_cu118
torchdiffeq=0.2.4=pypi_0
torchmetrics=1.4.0.post0=pypi_0
torchtriton=3.0.0=py312
torchvision=0.19.0=py312_cu118
tornado=6.3.3=py312h5eee18b_0
tqdm=4.66.4=py312he106c6f_0
truststore=0.8.0=py312h06a4308_0
typeguard=4.2.1=py312h06a4308_0
typing-extensions=4.11.0=py312h06a4308_0
typing_extensions=4.11.0=py312h06a4308_0
tzdata=2024.1=pypi_0
unicodedata2=15.1.0=py312h5eee18b_0
urllib3=2.1.0=py312h06a4308_1
wandb=0.18.1=pypi_0
webencodings=0.5.1=pypi_0
wheel=0.43.0=py312h06a4308_0
wrapt=1.16.0=pypi_0
xz=5.4.6=h5eee18b_1
yaml=0.2.5=h7b6447c_0
yaml-cpp=0.8.0=h6a678d5_1
zfp=1.0.0=h6a678d5_0
zlib=1.2.13=h5eee18b_1
zstandard=0.22.0=py312h2c38b39_0
zstd=1.5.5=hc292b87_2

**Comment by thanksforyourworks** [comment]
Hi, Thank you very much for your reply, I did have some trouble trying to train, for example always getting error messages in ninja package related operations, specifically in upfirdn2d. I tried several times to reconfigure the environment and paths without solving the problem all opened this help. I will try again based on the version you provided to rule out if it is a version conflict. Also, on line 122 in train_flow_matching.py, the project is unable to find the utils.save_image function, and I'm not sure if there's a misinterpretation in there. I tried to rewrite the function, but it didn't work. Would appreciate more information on being able to successfully run this GitHub version of the code. Thank you for your time!

**Comment by annegnx** [comment]
Hello, 
The `train_flow_matching.py` script should be fixed. Thanks for pointing this out.
